tests/test_spatial_filters.py

```diff
@@ -161,7 +186,98 @@ def test_init_with_invalid_spatial_filter(cli_runner, tmp_path):
+    crs = SPATIAL_FILTER_CRS["polygons"]
+
+    file_path = (tmp_path / "spatialfilter.txt").resolve()
+    file_path.write_text(f"{crs}\n\n{geom}\n", encoding="utf-8")
+
+    with data_archive("polygons-with-s2index") as repo1_path:
+        repo1_url = f"file://{repo1_path.resolve()}"
+        # Clone repo using spatial filter
+        repo2_path = tmp_path / "repo2"
+        r = cli_runner.invoke(
+            ["clone", repo1_url, repo2_path, f"--spatial-filter=@{file_path}"]
+        )
+        assert r.exit_code == 0, r.stderr
+
+        # The resulting repo has the spatial filter configured locally.
+        repo2 = KartRepo(repo2_path)
+        assert repo2.config["kart.spatialfilter.geometry"].startswith(
+            "POLYGON ((174.879 -37.8277,"
+        )
+        assert repo2.config["kart.spatialfilter.crs"] == crs
+
+        with repo2.working_copy.session() as sess:
+            assert H.row_count(sess, H.POLYGONS.LAYER) == 44
+
+        # However, the entire polygons layer was cloned.
+        # Spatial filters are currently only applied locally... all features are still present.
+        assert local_features(repo2.datasets()[H.POLYGONS.LAYER]) == H.POLYGONS.ROWCOUNT
+
+        # Unless you use an experimental environment variable:
+        # TODO: Always apply spatial filters during the clone, then clean up this test.
+        try:
+            os.environ["X_KART_SPATIAL_FILTERED_CLONE"] = "1"
+            repo3_path = tmp_path / "repo3"
+            r = cli_runner.invoke(
+                ["clone", repo1_url, repo3_path, f"--spatial-filter=@{file_path}"]
+            )
+            assert r.exit_code == 0, r.stderr
+
+            repo3 = KartRepo(repo3_path)
+            assert repo3.config["kart.spatialfilter.geometry"].startswith(
+                "POLYGON ((174.879 -37.8277,"
+            )
+            assert repo3.config["kart.spatialfilter.crs"] == crs
+            ds = repo3.datasets()[H.POLYGONS.LAYER]
+
+            local_feature_count = local_features(ds)
+            assert local_feature_count != H.POLYGONS.ROWCOUNT
+            assert local_feature_count == 52
+
+            with repo3.working_copy.session() as sess:
+                assert H.row_count(sess, H.POLYGONS.LAYER) == 44
+
+                # Inserting features that are in the dataset, but don't match the spatial filter,
+                # so they are not loaded locally nor written to the working copy.
+                for pk in H.POLYGONS.SAMPLE_PKS:
+                    if not is_local_feature(ds, pk):
+                        insert(sess, with_pk=pk, commit=False)
+
+            r = cli_runner.invoke(["-C", repo3_path, "status"])
+            assert r.exit_code == 0, r.stderr
+            # TODO - label pk-conflicts as being different to updates in kart status
+            assert "6 updates" in r.stdout
+            # All of the 6 featues that are conflicts / were "updated" in the WC have been loaded:
+            assert local_features(ds) == 58
```

Comment:
is there a (sane) way to test the number of objects passed to git to fetch on demand?